core/src/main/scala/cats/Foldable.scala

```diff
@@ -249,6 +249,38 @@ import Foldable.sentinel
    */
   def combineAll[A: Monoid](fa: F[A]): A = fold(fa)
 
+  /**
+   * Tear down a subset of this structure using a `A => Option[M]`.
+   *{{{
+   * scala> import cats.implicits._
+   * scala> val xs = List(1, 2, 3, 4)
+   * scala> def f(n: Int): Option[Int] = if (n % 2 == 0) Some(n) else None
+   * scala> xs.filterFold(f)
+   * res0: Int = 6
+   *}}}
+   */
+  def filterFold[A, M: Monoid](fa: F[A])(f: A ⇒ Option[M]): M = {
+    val m = Monoid[M]
+    foldLeft(fa, m.empty)((acc, a) ⇒ f(a) match {
+      case Some(x) ⇒ m.combine(x, acc)
```

Comment:
I think the order on combine is wrong. I think this won’t match collectFold for a non commutative Monoid.

Reply:
Yes, that's right, thanks, I will fix it soon :)